Memories/Memories/Views/CustomViews.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  CustomViews.swift
+//  Test
+//
+//  Created by Alexander Supe on 13.03.20.
+//  Copyright © 2020 Alexander Supe. All rights reserved.
+//
+
+import UIKit
+
+@IBDesignable class RoundButton: UIButton {
+    @IBInspectable var cornerRadius: CGFloat = 10 { didSet {
+        self.layer.cornerRadius = cornerRadius
+        }}
+    
+}
+
+class DetailView: UIView {
+    var memory: Memory? { didSet {
+        if let data = memory?.image {
+            image.image = UIImage(data: data)
+            addSubview(image)
+            image.clipsToBounds = true
+            image.translatesAutoresizingMaskIntoConstraints = false
+            image.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 5).isActive = true
+            image.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -5).isActive = true
+            image.topAnchor.constraint(equalTo: topAnchor, constant: 5).isActive = true
+            image.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -5).isActive = true
+            image.heightAnchor.constraint(equalToConstant: 250).isActive = true
+            image.contentMode = .scaleAspectFill
+        }
+        }}
+    private let image = UIImageView()
+}
```

Comment:
1. I would write this class using initializers and break up the work into discrete methods. 
2. I also prefer to use properties to create all my views. See the example below. 
3. `image` is confusing because this is not an image, it's an imageView. I like names that are verbose as to the thing, and what type it might be.

4. memory.image is not an image, it's a Data, and that should be called `imageData` Mixing and matching these objects is hard to follow when things are called the same thing but are not actual UIImage objects.


```swift
class DetailView: UIView {
    var memory: Memory? {
        didSet {
            if let imageData = memory?.image { // image data, not an image
                imageView.image = UIImage(data: imageData)
            }
        }
    }
        
    private var imageView: UIImageView = {
        let imageView = UIImageView()
        imageView.clipsToBounds = true
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.contentMode = .scaleAspectFill
        return imageView
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        addSubview(imageView)
        setUpConstraints()
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    private func setUpConstraints() {

        // Better to add constaints at the same time
        addConstraints([
            imageView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 5),
            imageView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -5),
            imageView.topAnchor.constraint(equalTo: topAnchor, constant: 5),
            imageView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -5),
            imageView.heightAnchor.constraint(equalToConstant: 250)
        ])
    }
}
```